vignettes/nactdt.Rmd

```diff
@@ -0,0 +1,220 @@
+ 
+ 
+### Prerequisite 
+
+Prior to calling the function,an input `date_source` object must be created for each source dataset using `admiral::date_source()`. Within this function, any company-specific subsetting can be passed to the `filter` argument. if numeric version of the dates will be needed,`admiral::convert_dtc_to_dt`, or if time part is needed, `admiral::convert_dtc_to_dtm` can be used as part of an expression for the `new_vars` argument. 
+
+```{r message=FALSE}
+cm_date <- date_source(
+  dataset_name = "cm",
+  filter = CMSCAT == "CHEMOTHERAPY" & CMCAT == "ON TREATMENT" & !is.na(CMSTDTC),
+  date = convert_dtc_to_dt(CMSTDTC)
+)
+
+pr_date <- date_source(
+  dataset_name = "pr",
+  filter = PRCAT == "CANCER RELATED" & PRSCAT == "ON TREATMENT" & !is.na(PRSTDTC),
+  date = convert_dtc_to_dt(PRSTDTC)
+)
+```
+
+## Multiple Sources - Derive New Anti-Cancer Start Date and Merge with `ADSL` {#nactdt}
+
+After completion of the prerequisite step, the new anti-cancer date can be derived while simultaneously adding it to `ADSL`.
+
+```{r message=FALSE}
+adsl <- admiral_adsl %>%
+  derive_var_extreme_dt(
+    new_var = NACTDT,
+    cm_date, pr_date,
+    source_datasets = list(
+      cm = cm,
+      pr = pr
+    ),
+    mode = "first"
+  )
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adsl,
+  display_vars = exprs(USUBJID, NACTDT),
+  filter = !is.na(NACTDT)
+)
+```
+## Derive Date Parameter {#par}
+
+For deriving parameter for new anti-cancer date from a single source, `admiral::derive_param_extreme_event()` may be used, please consult the `{admiralonco}` [Creating ADRS](adrs.html) for guidance on creating parameters and on the usage of `NACTDT` variable with examples. Additionally, subsection *Creating Your Own Time-to-Event Source Objects* in the `{admiralonco}` [Creating ADTTE](adtte.html) vignette may be referenced for case usage.
+
+
+As of `{admiral}` version 0.11.0, `admiral::derive_param_extreme_record()` is recommended to be used for deriving parameter based on the first or last observation from single and multiple sources. 
+
+Based on individual company standards, this parameter may be added to an event or date dataset such as `ADEVENT` or `ADDATES`, which are generally intermediate datasets to `ADTTE`. For demonstration purpose, a new anti-cancer date parameter will be merged into `ADRS`. A list of `record_source()` objects is expected and this will contain the sources of the dates and any company specific filtering.
+
+```{r, eval=TRUE, echo=FALSE}
+
+library(admiralonco)
+
+adrs <- derive_param_extreme_record(
+  dataset = admiral_adrs,
+  sources = list(
+    records_source(
+      dataset_name = "cm",
+      filter = CMSCAT == "CHEMOTHERAPY" & CMCAT == "ON TREATMENT" & !is.na(CMSTDTC),
+      new_vars = exprs(
+        ADT = convert_dtc_to_dt(CMSTDTC),
+        AVALC = CMTRT
```

Comment:
As we are creating AVALC by using CMTRT or PRTRT, I think we can mention as a text to say its individual company choice to create AVALC as per company standards.

Reply:
Might be a little redundant to use it again  since using company standard was mentioned in the beginning if the paragraph.